server/src/api/community.rs

```diff
@@ -472,7 +522,105 @@ impl Perform for Oper<EditCommunity> {
       res_sent.community.subscribed = None;
 
       ws.chatserver.do_send(SendCommunityRoomMessage {
-        op: UserOperation::EditCommunity,
+        op: UserOperation::DeleteCommunity,
+        response: res_sent,
+        community_id: data.edit_id,
+        my_id: ws.id,
+      });
+    }
+
+    Ok(res)
+  }
+}
+
+#[async_trait::async_trait(?Send)]
+impl Perform for Oper<RemoveCommunity> {
+  type Response = CommunityResponse;
+
+  async fn perform(
+    &self,
+    pool: &DbPool,
+    websocket_info: Option<WebsocketInfo>,
+  ) -> Result<CommunityResponse, LemmyError> {
+    let data: &RemoveCommunity = &self.data;
+
+    let claims = match Claims::decode(&data.auth) {
+      Ok(claims) => claims.claims,
+      Err(_e) => return Err(APIError::err("not_logged_in").into()),
+    };
+
+    let user_id = claims.id;
+
+    // Check for a site ban
+    let user = blocking(pool, move |conn| User_::read(conn, user_id)).await??;
+    if user.banned {
+      return Err(APIError::err("site_ban").into());
+    }
+
+    // Verify its an admin (only an admin can remove a community)
+    let admins: Vec<i32> = blocking(pool, move |conn| {
+      UserView::admins(conn).map(|v| v.into_iter().map(|a| a.id).collect())
+    })
+    .await??;
+    if !admins.contains(&user_id) {
+      return Err(APIError::err("not_an_admin").into());
+    }
+
+    // Do the remove
+    let edit_id = data.edit_id;
+    let removed = data.removed;
+    let updated_community = match blocking(pool, move |conn| {
+      Community::update_removed(conn, edit_id, removed)
+    })
+    .await?
+    {
+      Ok(community) => community,
+      Err(_e) => return Err(APIError::err("couldnt_update_community").into()),
+    };
+
+    // Mod tables
+    let expires = match data.expires {
+      Some(time) => Some(naive_from_unix(time)),
```

Comment:
What is the `expires` for? The community is only removed temporarily?

Reply:
Its a stub for now, (I was thinking about temp bans for reddit), but neither the front end sets them, nor the back end enforces them). Its okay to leave for another time when we fully want to work out temp bans.

Reply:
I'm not a fan of having this kind of unused code tbh. Better to remove it and implement it properly when we can actually test it.

Reply:
I'll put a todo in there for now. Otherwise it'd just be adding this exact code right back in when we eventually do this.